### PR TITLE
unit-test-transport

### PR DESCRIPTION
**Pull request by githublaohu**
unit-test-transport

### CODE REVIEW

File: test/src/transport/ClientRemotingProcessorTest.cpp [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+    request->SetBody(strData.c_str(), strData.size());
+    clientRemotingProcessor.resetOffset(request);
+
+    delete header;
+    delete request;
+}
+
+TEST(clientRemotingProcessorS, getConsumerRunningInfo) {
+    MockMQClientFactory *factory = new MockMQClientFactory("testClientId", 4, 3000, 4000, "a");
```
Comment (jonnxu) [inline test/src/transport/ClientRemotingProcessorTest.cpp]: factory and info not deleted,  please check the other TEST for not free the object, or use shared_ptr.   also i think you can use local object, no need with new

File: test/src/transport/ClientRemotingProcessorTest.cpp [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+using ::testing::InitGoogleMock;
+using ::testing::InitGoogleTest;
+using testing::Mock;
+using testing::Return;
+using testing::SetArgReferee;
+
+using Json::FastWriter;
+using Json::Value;
+
+using rocketmq::ClientRemotingProcessor;
```
Comment (jonnxu) [inline test/src/transport/ClientRemotingProcessorTest.cpp]: for use so many class under namespace rocketmq, i think you can use "using namespace rocketmq" replace of "using rocketmq::*"

File: test/src/transport/SocketUtilTest.cpp [hunk 3]
```diff
@@ -14,31 +14,34 @@
...
-using ::testing::InitGoogleTest;
+#include "gtest/gtest.h"
+
+#include "SocketUtil.h"
+
 using ::testing::InitGoogleMock;
+using ::testing::InitGoogleTest;
 using testing::Return;
 
-TEST(MQDecoderTest, messageProperties2String) {
```
Comment (jonnxu) [inline test/src/transport/SocketUtilTest.cpp]: why delete TEST(MQDecoderTest, messageProperties2String)?

File: test/src/transport/ClientRemotingProcessorTest.cpp (hunk 1 above)
Comment (githublaohu) [reply test/src/transport/ClientRemotingProcessorTest.cpp]: Mock::AllowLeak(factory);   Be equal to deleted

Comment (jonnxu) [reply test/src/transport/ClientRemotingProcessorTest.cpp]: but there are many point without Mock::AllowLeak

**Pull request merged**